cunumeric/deferred.py

```diff
@@ -95,24 +96,41 @@ def _prod(tpl: Sequence[int]) -> int:
 
 
 def auto_convert(
-    indices: Collection[int], keys: Sequence[str] = []
+    *thunk_params: str,
 ) -> Callable[[Callable[P, R]], Callable[P, R]]:
-    indices = set(indices)
+    """
+    Converts all named parameters to DeferredArrays.
+    """
+    keys: Set[str] = set(thunk_params)
```

Comment:
should we assert  no duplicate names are passed in `thunk_params`? Also I don't _think_ the explicit type annotation should be needed here since `thunk_params` has a type?

Reply:
57741e7bd460a423a1a51940882f98eaf90487ff